app/src/main/java/org/torproject/android/OrbotMainActivity.java

```diff
@@ -667,41 +688,44 @@ private synchronized void handleIntents() {
             return;
 
         switch (action) {
-            case INTENT_ACTION_REQUEST_HIDDEN_SERVICE:
-                final int hiddenServicePort = intent.getIntExtra("hs_port", -1);
-                final int hiddenServiceRemotePort = intent.getIntExtra("hs_onion_port", -1);
-                final String hiddenServiceName = intent.getStringExtra("hs_name");
-                final String backupToPackage = intent.getStringExtra("hs_backup_to_package");
-                final Boolean authCookie = intent.getBooleanExtra("hs_auth_cookie", false);
-                final Uri mKeyUri = intent.getData();
-
-                DialogInterface.OnClickListener dialogClickListener = (dialog, which) -> {
-                    if (which == DialogInterface.BUTTON_POSITIVE) {
-                        enableHiddenServicePort(
-                                hiddenServiceName, hiddenServicePort,
-                                hiddenServiceRemotePort, backupToPackage,
-                                mKeyUri, authCookie
-                        );
-                    }
-                };
-
-                String requestMsg = getString(R.string.hidden_service_request, String.valueOf(hiddenServicePort));
-                AlertDialog.Builder builder = new AlertDialog.Builder(this);
-                builder.setMessage(requestMsg).setPositiveButton("Allow", dialogClickListener)
-                        .setNegativeButton("Deny", dialogClickListener).show();
+            case INTENT_ACTION_REQUEST_V3_ONION_SERVICE:
+                final int v3LocalPort = intent.getIntExtra("localPort", -1);
+                final int v3onionPort = intent.getIntExtra("onionPort", v3LocalPort);
+                String name = intent.getStringExtra("name") ;
+                if (name == null) name = "v3" + v3LocalPort;
+                final String finalName = name;
+                new AlertDialog.Builder(this)
+                        .setMessage(getString(R.string.hidden_service_request, String.valueOf(v3LocalPort)))
+                        .setPositiveButton(R.string.allow, (d, w) -> enableV3OnionServiceOnV3(v3LocalPort, v3onionPort, finalName))
+                        .setNegativeButton(R.string.deny, (d, w) -> d.dismiss())
+                        .show();
+                return;
```

Comment:
I agree that configuring Onion Services should require user interaction by default.  Seems like this requires user interaction in order to start the hidden service. I think the right model would be to require user interaction to setup a new Onion Service or change an existing Onion Service config.  Outside of that, it seems to me the most useful would be to make the Onion Service start/stop just happen as part of the start/stop of Tor itself.